### PR TITLE
Publish sdks to prod improvements - approval required

### PR DESCRIPTION
**Pull request by pauld-msft**
You all likely received an email with "Run stage approval pending" in the subject. This is the pull request that will introduce this functionality to the publish to prod job. This allows us to have a manual approval before making SDKs available to production (since hopefully everything before this step will be automated eventually)

I manually denied the non-dry run job from running in this case.
image

### CODE REVIEW

File: vsts/pipelines/publishSdkToProd.yml
Comment (cormacpayne) [inline vsts/pipelines/publishSdkToProd.yml:49]: nit: it would be nice to include a comment on this line to call out that the `environment` property is needed to require approvals for this stage to run

**Review by cormacpayne — DISMISSED** [review]
:shipit:

**Comment by daniv-msft** [comment]
Looks good to me but, out of curiosity, could you please clarify where are defined the fact that we'll wait for an approver before pushing? Where is the list of approvers defined? I couldn't find it in the code change, but maybe I missed it. Or is it set up in the pipeline instead?

**Review by daniv-msft — APPROVED** [review]

File: vsts/scripts/publishSdksFromDevToProdStorageAccount.sh
Comment (william-msft) [inline vsts/scripts/publishSdksFromDevToProdStorageAccount.sh:7]: Curious why we're removing the `x`? The trace sometimes helps with debugging

Comment (pauld-msft) [reply vsts/scripts/publishSdksFromDevToProdStorageAccount.sh:7]: @william-msft I removed the x since in my experience with these jobs, the extra printing creates some noise in the job. Its a lot easier to review what is going to change during the dry-run (which we will be looking at before manually approving), and in this script we are quite verbose with our printing, to the point where we can quickly determine what step/blob it may have failed on.

Comment (william-msft) [reply vsts/scripts/publishSdksFromDevToProdStorageAccount.sh:7]: Sounds good

**Review by william-msft — APPROVED** [review]

**Pull request merged**
